### PR TITLE
Migrate from TSLint to ESLint

### PR DESCRIPTION
**Pull request by Maks1mS**
TSLint has been deprecated in favor of ESLint https://github.com/palantir/tslint/issues/4534.

### CODE REVIEW

**Comment by Maks1mS** [comment]
In the current configuration, eslint reports 59 problems (54 errors, 5 warnings).
Need to change the configuration or fix problems? Most of them concern [prefer-const](https://eslint.org/docs/latest/rules/prefer-const).

**Comment by mre** [comment]
Good idea to move to eslint!

Perhaps we can fix as many problems as possible on this branch if it's just about simple fixes. If we hit a big blocker, we can [exclude those lines](https://stackoverflow.com/a/29592334) individually. We can then fix them later without blocking the migration to eslint. What I like about this approach is that it's explicit: we can grep for `eslint-disable-next-line` in the codebase and fix a few lints whenever we have time without forgetting about them. 😉

**Comment by mre** [comment]
Huh! Either eslint was extremely fast (2 seconds) or it didn't run. There isn't much in the Github Actions output. 🤔 Does it pass locally?

**Comment by Maks1mS** [comment]
> Huh! Either eslint was extremely fast (2 seconds) or it didn't run. There isn't much in the Github Actions output. thinking Does it pass locally?

In this PR, mooyoul/tslint-actions is executed within 2 seconds (because tslint is removed).
In the future, PR will `run npm run lint`, which runs `eslint src` (we don't need [run-eslint](https://github.com/marketplace/actions/run-eslint)).

**Comment by mre** [comment]
Thanks for the clarification. Would it make sense to remove `mooyoul/tslint-actions` in this PR since it's unused?

**Comment by Maks1mS** [comment]
> Thanks for the clarification. Would it make sense to remove `mooyoul/tslint-actions` in this PR since it's unused?

It has already been removed in this PR. It starts because it is defined in master branch.

https://github.com/mre/vscode-snippet/pull/210/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fL24-L31

**Comment by mre** [comment]
Ah, got it. Looks good then. 👍

File: package.json [hunk 1]
```diff
@@ -240,6 +244,10 @@
     "eslint": "^8.33.0",
     "glob": "^8.1.0",
     "mocha": "^10.2.0",
+    "rimraf": "^4.1.2",
+    "rollup": "^3.14.0",
+    "rollup-plugin-node-externals": "^5.1.2",
+    "tslint": "^6.1.3",
```
Comment (Maks1mS) [inline package.json:250]: Whoops! 
It shouldn't have been here.

Comment (Maks1mS) [reply package.json:250]: it looks like I made a mistake in updating from master.

Comment (Maks1mS) [reply package.json:250]: #212

**Pull request merged**